### ISSUE TITLE
Question about 'target' and 'time_varying_unknown_reals' in TimeSeriesDataSet

### ISSUE THREAD

**Issue by havanoname** [jdb78/pytorch-forecasting#1437]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version: 2.0
- Python version: 3.10.9
- Operating System: Win11

### Expected behavior
I made some changes to the code in the [tutorial](https://pytorch-forecasting.readthedocs.io/en/stable/tutorials/building.html):

```
from typing import Dict
import numpy as np
import pandas as pd
from pytorch_forecasting import TimeSeriesDataSet
from sklearn.preprocessing import StandardScaler

v1 = np.arange(10)
v2 = np.arange(10)*2
test_data = pd.DataFrame(
    dict(
        value1=v1,
        value2=v2,
        value3=v1 + v2,
        group=np.repeat(np.arange(2), 5),
        time_idx=np.tile(np.arange(5), 2),
    )
)
transfer = StandardScaler()
data_new = transfer.fit_transform(v1.reshape(-1,1))
test_data['norm_value'] = data_new
print(test_data)

# create the dataset from the pandas dataframe
dataset = TimeSeriesDataSet(
    test_data,
    group_ids=["group"],
    target="value3",
    time_idx="time_idx",
    min_encoder_length=2,
    max_encoder_length=2,
    min_prediction_length=2,
    max_prediction_length=2,
    time_varying_unknown_reals=["value1", "value2", "value3"],
)
# convert the dataset to a dataloader
dataloader = dataset.to_dataloader(batch_size=4, train=False)

# and load the first batch
x, y = next(iter(dataloader))
print("x =", x)
print("\ny =", y)
```
the output of the program is listed at the end, I also posted the specific data on [StackOverflow](https://stackoverflow.com/questions/77440127/pytorch-forecasting-understanding-target-and-time-varying-unknown-reals-in), 

In my opinion, this is an autoregressive problem with covariates: $(v1_{t-1}, v2_{t-1}, v3_{t-1}),(v1_t, v2_t, v3_t) \rightarrow v3_{t+1},v3_{t+2}$. If $v3$ is removed in `time_varying_unknown_reals`, it is a prediction problem: $(v1_{t-1}, v2_{t-1}),(v1_t, v2_t) \rightarrow v3_{t+1},v3_{t+2}$

So I think the result I should get is: 
```
...
'encoder_cont': tensor([[[-1.5667, -1.5667,  0.0000],
         [-1.2185, -1.2185,  3.0000]],

        [[-1.2185, -1.2185,  3.0000],
         [-0.8704, -0.8704,  6.0000]],

        [[ 0.1741,  0.1741,  6.0000],
         [ 0.5222,  0.5222,  9.0000]],

        [[ 0.5222,  0.5222,  9.0000],
         [ 0.8704,  0.8704,  12.0000]]])
...
y = (tensor([[3, 6],
        [6, 9],
        [9, 12],
        [12, 15]]), None)
```
The first and second columns are normalized to `value1` and `value2`, the third column `value3` is the target, so there is no standardization.

### Actual behavior
However, result was: 
```
...
'encoder_cont': tensor([[[-1.5667, -1.5667,  0.0000],
         [-1.2185, -1.2185,  1.0000]],

        [[-1.2185, -1.2185,  1.0000],
         [-0.8704, -0.8704,  2.0000]],

        [[ 0.1741,  0.1741,  5.0000],
         [ 0.5222,  0.5222,  6.0000]],

        [[ 0.5222,  0.5222,  6.0000],
         [ 0.8704,  0.8704,  7.0000]]])
...
y = (tensor([[2, 3],
        [3, 4],
        [7, 8],
        [8, 9]]), None)
```
The third column is not `value3`, but a table index. The output `y` also is table index.

I don't know why, the third column is the index value instead of `value3`. Do I have any misunderstandings about `TimeSeriesDataSet`?

### The output of the code

```
   value1  value2  value3  group  time_idx  norm_value
0       0       0       0      0         0    -1.566699
1       1       2       3      0         1    -1.218544
2       2       4       6      0         2    -0.870388
3       3       6       9      0         3    -0.522233
4       4       8      12      0         4    -0.174078
5       5      10      15      1         0     0.174078
6       6      12      18      1         1     0.522233
7       7      14      21      1         2     0.870388
8       8      16      24      1         3     1.218544
9       9      18      27      1         4     1.566699
x = {'encoder_cat': tensor([], size=(4, 2, 0), dtype=torch.int64), 'encoder_cont': tensor([[[-1.5667, -1.5667,  0.0000],
         [-1.2185, -1.2185,  1.0000]],

        [[-1.2185, -1.2185,  1.0000],
         [-0.8704, -0.8704,  2.0000]],

        [[ 0.1741,  0.1741,  5.0000],
         [ 0.5222,  0.5222,  6.0000]],

        [[ 0.5222,  0.5222,  6.0000],
         [ 0.8704,  0.8704,  7.0000]]]), 'encoder_target': tensor([[0, 1],
        [1, 2],
        [5, 6],
        [6, 7]]), 'encoder_lengths': tensor([2, 2, 2, 2]), 'decoder_cat': tensor([], size=(4, 2, 0), dtype=torch.int64), 'decoder_cont': tensor([[[-0.8704, -0.8704,  2.0000],
         [-0.5222, -0.5222,  3.0000]],

        [[-0.5222, -0.5222,  3.0000],
         [-0.1741, -0.1741,  4.0000]],

        [[ 0.8704,  0.8704,  7.0000],
         [ 1.2185,  1.2185,  8.0000]],

        [[ 1.2185,  1.2185,  8.0000],
         [ 1.5667,  1.5667,  9.0000]]]), 'decoder_target': tensor([[2, 3],
        [3, 4],
        [7, 8],
        [8, 9]]), 'decoder_lengths': tensor([2, 2, 2, 2]), 'decoder_time_idx': tensor([[2, 3],
        [3, 4],
        [2, 3],
        [3, 4]]), 'groups': tensor([[0],
        [0],
        [1],
        [1]]), 'target_scale': tensor([[0., 0.],
        [0., 0.],
        [0., 0.],
        [0., 0.]])}

y = (tensor([[2, 3],
        [3, 4],
        [7, 8],
        [8, 9]]), None)
```
